tests/tracer/test_agent.py

```diff
@@ -23,15 +23,50 @@ def test_stats_port(monkeypatch):
     assert agent.get_stats_port() == 1235
 
 
-def test_trace_url(monkeypatch):
+def test_trace_url(monkeypatch, mocker):
+    mock_exists = mocker.patch("os.path.exists")
+
+    # with nothing set by user, and the default UDS available, we choose UDS
+    mock_exists.return_value = True
+    assert agent.get_trace_url() == "unix:///var/run/datadog/apm.socket"
+    mock_exists.assert_called_once_with("/var/run/datadog/apm.socket")
+    mock_exists.reset_mock()
+
+    # with nothing set by user, and the default UDS unavailable, we choose default http address
+    mock_exists.return_value = False
     assert agent.get_trace_url() == "http://localhost:8126"
+    mock_exists.assert_called_once_with("/var/run/datadog/apm.socket")
+    mock_exists.reset_mock()
+
+    # with port and host set by user, and default UDS unavailable, we choose user settings
     monkeypatch.setenv("DD_TRACE_AGENT_PORT", "1235")
     assert agent.get_trace_url() == "http://localhost:1235"
+    mock_exists.assert_not_called()
+
```

Comment:
```suggestion
```

You can set both env variables together and then call `get_trace_url()` once.

Reply:
optional suggested change

Reply:
I thought it'd be better to leave them separate because that way we test just setting one, and also test setting both in case there's an edge case.